lib/install/opsuser/opsuser.go

```diff
@@ -47,26 +47,29 @@ type resourceDesc struct {
 	ref   types.ManagedObjectReference
 }
 
-func NewRBACManager(ctx context.Context, client *vim25.Client, session *session.Session, rbacConfig *rbac.Config, configSpec *config.VirtualContainerHostConfigSpec) *RBACManager {
+func NewRBACManager(ctx context.Context, session *session.Session, rbacConfig *rbac.Config, configSpec *config.VirtualContainerHostConfigSpec) *RBACManager {
 	rbacMgr := &RBACManager{
 		configSpec: configSpec,
 		session:    session,
-		client:     client,
 	}
-	am := rbac.NewAuthzManager(ctx, client)
+	if session != nil {
```

Comment:
What happens if `session` is nil?

Reply:
The `client` field in the `RBACManager` struct would be nil, and calls that rely on the client field will fail. There are limited calls to `NewRBACManager` in our code, all of which pass in a valid session.

Reply:
I'll ask the same question but in a different way - if the `RBACManager` isn't valid without a valid session then why aren't we returning nil or an `error`?
Current approach results in an undocumented obligation on the caller across package boundaries.